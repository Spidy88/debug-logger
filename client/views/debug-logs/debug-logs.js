import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import LogRow from './log-row';
import LogControls from './log-controls';
import EmptyFeedback from '../../components/empty-feedback';
import ServerStatus from '../../components/server-status';
import {
    Pane,
    Table
} from 'evergreen-ui';

class DebugLogs extends Component {
    static propTypes = {
        logs: PropTypes.array,
        status: PropTypes.string
    };

    static defaultProps = {
        logs: []
    };

    constructor(props) {
        super(props);

        this.renderEmptyState = this.renderEmptyState.bind(this);
        this.renderServerStatus = this.renderServerStatus.bind(this);
        this.render = this.render.bind(this);
    }

    renderEmptyState() {
        if( this.props.logs.length ) {
            return null;
        }

        let iconProps = {
            icon: 'barber',
            message: 'Waiting for logs',
            subMessage: 'Kick back and relax'
        };

        // Render a waiting for logs feedback
        return (
            <EmptyFeedback {...iconProps} />
        );
    }

    renderServerStatus() {
        if( !this.props.status ) {
            return null;
        }

        return (
            <ServerStatus status={this.props.status} />
        );
    }

    render() {
        return (
            <React.Fragment>
                <Pane
                    background="tint2"
                    paddingX="0.75rem"
                    display="flex"
                    position="fixed"
                    width="100%"
                    top={0}
                    elevation={1}>

                    <LogControls />
                </Pane>

                <Box marginTop="calc(40px + 2rem)">
                    { this.renderServerStatus() }

                    { this.renderEmptyState() }

                    <Table>
                        <Table.Body>
                            {
                                this.props.logs.map((log) => (
                                    <LogRow key={log.messageId} log={log} />
                                ))
                            }
                        </Table.Body>
                    </Table>
                </Box>
            </React.Fragment>
        );
    }
}

export default DebugLogs;
