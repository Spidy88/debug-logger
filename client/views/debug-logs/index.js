import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LogRow from './log-row';
import LogControls from './log-controls';
import {
    Pane,
    Table
} from 'evergreen-ui';

class DebugLogs extends Component {
    static propTypes = {
        logs: PropTypes.array
    };

    static defaultProps = {
        logs: []
    };

    constructor(props) {
        super(props);

        this.render = this.render.bind(this);
    }

    render() {
        return (
            <React.Fragment>
                <Pane
                    background="tint2"
                    paddingX="0.75rem"
                    display="flex"
                    position="fixed"
                    width="100%"
                    top={0}
                    elevation={1}>

                    <LogControls />
                </Pane>

                <Table>
                    <Table.Body>
                        {
                            this.props.logs.map((log) => (
                                <LogRow key={log.messageId} log={log} />
                            ))
                        }
                    </Table.Body>
                </Table>
            </React.Fragment>
        );
    }
}

export default DebugLogs;
