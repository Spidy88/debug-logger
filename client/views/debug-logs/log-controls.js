import React, { PureComponent } from 'react';
import Box from 'ui-box';

import {
    SearchInput,
    SegmentedControl
} from 'evergreen-ui';

const LiveOption = {
    label: 'Live',
    value: 'live'
};

const PauseOption = {
    label: 'Pause',
    value: 'pause'
};

class LogControls extends PureComponent {
    constructor(props) {
        super(props);

        this.render = this.render.bind(this);
    }

    render() {
        return (
            <React.Fragment>
                <Box marginX="0.75rem" marginY="1rem" background="white">
                    <SegmentedControl
                        options={[LiveOption, PauseOption]}
                        defaultValue={LiveOption.value}
                        height={40}
                        flexShrink={1}
                        flexGrow={0}
                        width="10rem" />
                </Box>

                <Box marginX="0.75rem" marginY="1rem" width="100%">
                    <SearchInput
                        height={40}
                        flexGrow={1}
                        width="100%"
                        placeholder="Type to search..." />
                </Box>
            </React.Fragment>
        );
    }
}

export default LogControls;