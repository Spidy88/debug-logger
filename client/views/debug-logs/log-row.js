import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Box from 'ui-box';
import moment from 'moment';
import {
    Table,
    defaultTheme
} from 'evergreen-ui';

const eventTextProps = {
    fontWeight: 500,
    textAlign: 'left'
};

const dateTextProps = {
    textAlign: 'right'
};

class LogRow extends PureComponent {
    static propTypes = {
        log: PropTypes.shape({
            messageId: PropTypes.string.isRequired,
            sentAt: PropTypes.number.isRequired,
            event: PropTypes.string
        }).isRequired
    };

    constructor(props) {
        super(props);

        this.formattedLogDate = this.formattedLogDate.bind(this);
        this.render = this.render.bind(this);
    }

    formattedLogDate() {
        let { log } = this.props;
        return moment(log.sentAt).format('YYYY/MM/DD HH:mm:ss');
    }

    render() {
        let {
            log,
            ...props
        } = this.props;

        return (
            <Table.Row paddingX="0.75em" {...props}>
                <Table.Cell
                    width={40}
                    flexBasis="auto"
                    flex={0}
                    marginX="0.75em">

                    <Box
                        is="svg"
                        viewBox="0 0 24 24"
                        style={ { fill: defaultTheme.fills.solid.blue.backgroundColor } }
                        width={24}
                        height={24}>

                        <path d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10-10-4.486-10-10 4.486-10 10-10zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm6.25 8.891l-1.421-1.409-6.105 6.218-3.078-2.937-1.396 1.436 4.5 4.319 7.5-7.627z"/>
                    </Box>
                </Table.Cell>

                <Table.TextCell
                    width="8em"
                    flexBasis="auto"
                    flex={0}
                    textTransform="uppercase"
                    fontWeight={300}
                    test-id="logType">

                    { log.type }
                </Table.TextCell>

                <Table.TextCell
                    textProps={eventTextProps}
                    test-id="logMessage">

                    { log.event || log.messageId }
                </Table.TextCell>

                <Table.TextCell
                    width="10em"
                    marginX="0.75em"
                    flexBasis="auto"
                    flex={0}
                    textProps={dateTextProps}
                    test-id="logDate">

                    { this.formattedLogDate() }
                </Table.TextCell>
            </Table.Row>
        );
    }
}

export default LogRow;