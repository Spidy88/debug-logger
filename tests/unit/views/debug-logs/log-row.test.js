import React from 'react';
import renderer from 'react-test-renderer';
import timezoneMock from 'timezone-mock';
import { render } from 'enzyme';
import { LogRow } from '../../../../client/views/debug-logs';

describe('<LogRow>', () => {
    const logDateFormat = /\d{4}\/\d{2}\/\d{2} \d{2}:\d{2}:\d{2}/;

    beforeAll(() => {
        timezoneMock.register('UTC');
    });

    afterAll(() => {
        timezoneMock.unregister();
    });

    it('should render the log type and event', () => {
        const testLog = {
            messageId: '123abc',
            type: 'track',
            event: 'Add to Cart',
            sentAt: 1545345141076
        };
        const logRow = render(<LogRow log={testLog} />);

        expect(logRow.find('[test-id=logType]').text()).toEqual(testLog.type);
        expect(logRow.find('[test-id=logMessage]').text()).toEqual(testLog.event);
        expect(logRow.find('[test-id=logDate]').text()).toMatch(logDateFormat);
    });

    it('should render the log type and id if no event is provided', () => {
        const testLog = {
            messageId: 'abc123',
            type: 'identify',
            sentAt: 1545345141076
        };
        const logRow = render(<LogRow log={testLog} />);

        expect(logRow.find('[test-id=logType]').text()).toEqual(testLog.type);
        expect(logRow.find('[test-id=logMessage]').text()).toEqual(testLog.messageId);
        expect(logRow.find('[test-id=logDate]').text()).toMatch(logDateFormat);
    });

    it('should match its snapshot', () => {
        const testLog = {
            messageId: '123abc',
            type: 'track',
            event: 'Add to Cart',
            sentAt: 1545345141076
        };
        const tree = renderer.create(<LogRow log={testLog} />);

        expect(tree.toJSON()).toMatchSnapshot();
    });
});