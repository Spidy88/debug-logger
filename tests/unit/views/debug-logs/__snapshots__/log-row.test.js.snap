// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogRow> should match its snapshot 1`] = `
<div
  className="css-z3ro6d ðŸ“¦b-btm_1px-solid-EDF0F2 ðŸ“¦dspl_flex ðŸ“¦h_48px ðŸ“¦pl_0-75em ðŸ“¦pr_0-75em ðŸ“¦box-szg_border-box css-nil"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <div
    className="css-saeo60 ðŸ“¦pl_12px ðŸ“¦pr_12px ðŸ“¦box-szg_border-box ðŸ“¦flx_0 ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-srnk_0 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦w_40px ðŸ“¦flx-basis_auto ðŸ“¦ml_0-75em ðŸ“¦mr_0-75em css-nil"
    onKeyDown={[Function]}
  >
    <svg
      className="ðŸ“¦w_24px ðŸ“¦h_24px ðŸ“¦box-szg_border-box"
      style={
        Object {
          "fill": "#1070ca",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10-10-4.486-10-10 4.486-10 10-10zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm6.25 8.891l-1.421-1.409-6.105 6.218-3.078-2.937-1.396 1.436 4.5 4.319 7.5-7.627z"
      />
    </svg>
  </div>
  <div
    className="css-saeo60 ðŸ“¦pl_12px ðŸ“¦pr_12px ðŸ“¦box-szg_border-box ðŸ“¦flx_0 ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-srnk_0 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦w_8em ðŸ“¦flx-basis_auto ðŸ“¦txt-trns_uppercase ðŸ“¦f-wght_300 css-nil"
    onKeyDown={[Function]}
    test-id="logType"
  >
    <span
      className="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦flx_1 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦txt-ovrf_ellipsis ðŸ“¦wht-spc_nowrap ðŸ“¦box-szg_border-box"
    >
      track
    </span>
  </div>
  <div
    className="css-saeo60 ðŸ“¦pl_12px ðŸ“¦pr_12px ðŸ“¦box-szg_border-box ðŸ“¦flx_1 ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-srnk_0 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden css-nil"
    onKeyDown={[Function]}
    test-id="logMessage"
  >
    <span
      className="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦flx_1 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦txt-ovrf_ellipsis ðŸ“¦wht-spc_nowrap ðŸ“¦txt-algn_left ðŸ“¦box-szg_border-box"
    >
      Add to Cart
    </span>
  </div>
  <div
    className="css-saeo60 ðŸ“¦pl_12px ðŸ“¦pr_12px ðŸ“¦box-szg_border-box ðŸ“¦flx_0 ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-srnk_0 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦w_10em ðŸ“¦ml_0-75em ðŸ“¦mr_0-75em ðŸ“¦flx-basis_auto css-nil"
    onKeyDown={[Function]}
    test-id="logDate"
  >
    <span
      className="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦flx_1 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦txt-ovrf_ellipsis ðŸ“¦wht-spc_nowrap ðŸ“¦txt-algn_right ðŸ“¦box-szg_border-box"
    >
      2018/12/20 22:32:21
    </span>
  </div>
</div>
`;
